catroid/src/main/java/org/catrobat/catroid/utils/storagestuff/ProjectsStorageUtils.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.utils.storagestuff;
+
+import android.util.Log;
+
+import org.catrobat.catroid.common.FlavoredConstants;
+import org.catrobat.catroid.io.StorageOperations;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+import static org.catrobat.catroid.common.FlavoredConstants.EXTERNAL_STORAGE_ROOT_DIRECTORY;
+import static org.catrobat.catroid.common.FlavoredConstants.POCKET_CODE_EXTERNAL_STORAGE_FOLDER_NAME;
+import static org.catrobat.catroid.io.StorageOperations.getUniqueFile;
+import static org.catrobat.catroid.io.StorageOperations.transferData;
+
+public class ProjectsStorageUtils {
+
+	public static final String TAG = ProjectsStorageUtils.class.getSimpleName();
+	private static final String POCKET_CODE_INTERNAL_STORAGE_FOLDER_NAME = "files";
+	private static final String BACKPACK_FOLDER_NAME = "backpack";
+
+	private void copyFolderRecursive(File srcDir, File dstDir) throws IOException {
```

Comment:
please split, have one function that just deals with getting a list of projects that are to be copied and deal with unique name creation, and then call copyDir (or whatever suitable method there is in StorageHandling class) to do the actual copying for each project separately.